File: Drapeau.c
//Exercice C pour Projet de la vie courante
//Poste de contrôle de la plage qui contrôle l'état de la mer et qui hisse le drapeau en fonction du temps et l'état de la mer

//On inclue les différentes bibliothèques utilisés
#include <stdio.h>
#include <string.h>


//On créé la focntion principale
int main(void)
{

//On crée une variable char (caractère) qui alloue 5 de mémoire (des bytes)
    char a[5];


//On crée un tableau en 2D qui comporte 3 ligne et 3 colonnes.
//* sert à savoir quand le caractère devient Null.
    char *categorie[3][3] = 
    {
        {"Hisser le Drapeau vert ! Tout va bien pour les vacanciers.", "Plate", "Calme"},
        {"Hisser le Drapeau Orange ! Faite attention aux vacanciers.", "Agité", "Mauvais"},
        {"Hisser le Drapeau Rouge ! Personne de va se baigner !", "Dangeureuse", "Orage"}
    };


//On affiche le message si dessous grâce à la commande printf
///Scanf sert à assigner à une variable qui est (a) la valeur entré au clavier
    printf("Ici le poste de contrôle de la plage, comment est la mer ce matin pour les vacanciers (Plate, Agité, Dangeureuse, Calme, Mauvais ou Orage) ?\n");
    scanf("%s", a);


//On créé un boucle for qui va comparer les 2 derniers termes des trois lignes du tableau.
//A la variable a.
    for (int i = 0; i < 3; i++)
    {

//On créé un condition if qui va nou permettre de compararer grâce à strcmp les deux derniers termes de chaque ligne du tableau.
//strcmp est utliser grâce à la bibliothèque <string.h>
//Ne mettez pas d'autres mots que ceux du tableau sinon gare aux vacanciers !
        if (strcmp(a, categorie[i][1]) == 0 || strcmp(a, categorie[i][2]) == 0)
        {
            
//On affiche la valeur dans le terminal
            printf("%s\n", categorie[i][0]);


//On mets fin au programme principal
            return 0;
        }

    }
//On affiche le message si on entre pas les bons mots.
    printf("Vous mettez la vie des vacanciers en jeu !\n");
}

File: Drapeau.py
# Exercice Python pour Projet de la vie courante
# Poste de contrôle de la plage qui voit l'état de la mer et qui hisse le drapeau en fonction du temps et l'état de la mer

# On importe la fonction exit
import sys

# On créé un tableau 2D qui s'appelle categorie et qui comporte 3 lignes et 3 colonnes
categorie = [
    ["Hisser le Drapeau vert ! Tout va bien pour les vacanciers.", "Plate", "Calme"],
    ["Hisser le Drapeau Orange ! Faite attention aux vacanciers.", "Agité", "Forte"],
    ["Hisser le Drapeau Rouge ! Personne ne va se baigner !", "Dangeureuse", "Orage"]
]

# On affiche le message suivant et on assigne a (a) la réponse entré dans le terminal grâce à la fonction input.
a = input("Ici le poste de contrôle de la plage, comment est la mer ce matin pour les vacanciers (Plate, Agité, Dangeureuse, Calme, Forte ou Orage) ?\n")

# On créé une boucle for qui va initialiser 3 fois en partant de 0
for i in range(3):

    # On créé un condition if qui va nous permettre de compararer si la variable a est égal aux deux derniers termes de chaque ligne i qu'on compare.
    if a == categorie[i][1] or a == categorie[i][2]:

        # On affiche le premier terme de la ligne 0 du tableau.
        print(categorie[i][0])

        # On sort du programme principal
        sys.exit(0)

# On affiche ce message si on écrit pas l'un des mots dans les 2 dernières termes dans chaque ligne du tableau.
print("Vous mettez la vie des vacanciers en jeu !\n")

